From 1f7a61b48ec3c6ac8cbefc1c4a49e40477f3ae00 Mon Sep 17 00:00:00 2001
From: Summer Qin <summer.qin@arm.com>
Date: Fri, 17 Jun 2022 15:42:55 +0800
Subject: [PATCH 3/4] Fix 2 issues in FF test

- Disable secure test when testing version policy. Otherwise, the
received signal is not matched.
- In test 27, set wrong parameter and let SPM return expected
PSA_ERROR_PROGRAMMER_ERROR, instead of making test to directly
return the error code.

Signed-off-by: Summer Qin <summer.qin@arm.com>
---
 api-tests/ff/ipc/test_i005/test_entry_i005.c | 2 +-
 api-tests/ff/ipc/test_i006/test_entry_i006.c | 2 +-
 api-tests/ff/ipc/test_i007/test_entry_i007.c | 2 +-
 api-tests/ff/ipc/test_i010/test_entry_i010.c | 2 +-
 api-tests/ff/ipc/test_i011/test_entry_i011.c | 2 +-
 api-tests/ff/ipc/test_i027/test_i027.c       | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/api-tests/ff/ipc/test_i005/test_entry_i005.c b/api-tests/ff/ipc/test_i005/test_entry_i005.c
index 099d590..78c0ad5 100644
--- a/api-tests/ff/ipc/test_i005/test_entry_i005.c
+++ b/api-tests/ff/ipc/test_i005/test_entry_i005.c
@@ -40,7 +40,7 @@ void test_entry(val_api_t *val_api, psa_api_t *psa_api)
     }
 
     /* Execute list of tests available in test[num]_client_tests_list from Non-secure side*/
-    status = val->execute_non_secure_tests(TEST_NUM, test_i005_client_tests_list, TRUE);
+    status = val->execute_non_secure_tests(TEST_NUM, test_i005_client_tests_list, FALSE);
     if (VAL_ERROR(status))
     {
         goto test_exit;
diff --git a/api-tests/ff/ipc/test_i006/test_entry_i006.c b/api-tests/ff/ipc/test_i006/test_entry_i006.c
index 13f0b7d..c9b2996 100644
--- a/api-tests/ff/ipc/test_i006/test_entry_i006.c
+++ b/api-tests/ff/ipc/test_i006/test_entry_i006.c
@@ -40,7 +40,7 @@ void test_entry(val_api_t *val_api, psa_api_t *psa_api)
     }
 
     /* Execute list of tests available in test[num]_client_tests_list from Non-secure side*/
-    status = val->execute_non_secure_tests(TEST_NUM, test_i006_client_tests_list, TRUE);
+    status = val->execute_non_secure_tests(TEST_NUM, test_i006_client_tests_list, FALSE);
     if (VAL_ERROR(status))
     {
         goto test_exit;
diff --git a/api-tests/ff/ipc/test_i007/test_entry_i007.c b/api-tests/ff/ipc/test_i007/test_entry_i007.c
index 86ea352..e133488 100644
--- a/api-tests/ff/ipc/test_i007/test_entry_i007.c
+++ b/api-tests/ff/ipc/test_i007/test_entry_i007.c
@@ -40,7 +40,7 @@ void test_entry(val_api_t *val_api, psa_api_t *psa_api)
     }
 
     /* Execute list of tests available in test[num]_client_tests_list from Non-secure side*/
-    status = val->execute_non_secure_tests(TEST_NUM, test_i007_client_tests_list, TRUE);
+    status = val->execute_non_secure_tests(TEST_NUM, test_i007_client_tests_list, FALSE);
     if (VAL_ERROR(status))
     {
         goto test_exit;
diff --git a/api-tests/ff/ipc/test_i010/test_entry_i010.c b/api-tests/ff/ipc/test_i010/test_entry_i010.c
index e0e72ef..7d30848 100644
--- a/api-tests/ff/ipc/test_i010/test_entry_i010.c
+++ b/api-tests/ff/ipc/test_i010/test_entry_i010.c
@@ -40,7 +40,7 @@ void test_entry(val_api_t *val_api, psa_api_t *psa_api)
     }
 
     /* Execute list of tests available in test[num]_client_tests_list from Non-secure side*/
-    status = val->execute_non_secure_tests(TEST_NUM, test_i010_client_tests_list, TRUE);
+    status = val->execute_non_secure_tests(TEST_NUM, test_i010_client_tests_list, FALSE);
     if (VAL_ERROR(status))
     {
         goto test_exit;
diff --git a/api-tests/ff/ipc/test_i011/test_entry_i011.c b/api-tests/ff/ipc/test_i011/test_entry_i011.c
index a1ca9c7..c694c8f 100644
--- a/api-tests/ff/ipc/test_i011/test_entry_i011.c
+++ b/api-tests/ff/ipc/test_i011/test_entry_i011.c
@@ -40,7 +40,7 @@ void test_entry(val_api_t *val_api, psa_api_t *psa_api)
     }
 
     /* Execute list of tests available in test[num]_client_tests_list from Non-secure side*/
-    status = val->execute_non_secure_tests(TEST_NUM, test_i011_client_tests_list, TRUE);
+    status = val->execute_non_secure_tests(TEST_NUM, test_i011_client_tests_list, FALSE);
     if (VAL_ERROR(status))
     {
         goto test_exit;
diff --git a/api-tests/ff/ipc/test_i027/test_i027.c b/api-tests/ff/ipc/test_i027/test_i027.c
index 9b74a58..93c766c 100644
--- a/api-tests/ff/ipc/test_i027/test_i027.c
+++ b/api-tests/ff/ipc/test_i027/test_i027.c
@@ -78,7 +78,7 @@ int32_t client_test_psa_drop_connection(caller_security_t caller)
        return VAL_STATUS_ERROR;
    }
 
-   status_of_call =  psa->call(handle, PSA_IPC_CALL, NULL, 0, NULL, 0);
+   status_of_call =  psa->call(handle, PSA_IPC_CALL, NULL, PSA_MAX_IOVEC, NULL, PSA_MAX_IOVEC);
 
    /*
     * If the caller is in the NSPE, it is IMPLEMENTATION DEFINED whether
-- 
2.17.1

